### DISCUSSION TITLE
Reading local files using edge functions

### DISCUSSION THREAD

**Discussion by pnewell** [supabase/supabase#27287] [Questions]
I am trying to load a WASM file in my edge function folder but I have yet to figure out the right way to handle it.

```
supabase
    functions
        send-message
            index.ts
```

I have tried placing the main.wasm file in functions and send-message folders but always still get the following error:
InvalidWorkerResponse: event loop error: NotFound: path not found functions/send-message/main.wasm: readfile './functions/send-message/main.wasm'

Here is my index.ts file:
```
Deno.serve(async (req) => {
  let file = Deno.readFile("./functions/send-message/main.wasm");

  const importObject = { imports: { sendMessage: (arg) => console.log(arg) } };

  WebAssembly.instantiateStreaming(file, importObject).then(
    (obj) => { 
      return new Response(
        obj.instance.exports.sendMessage(),
        { headers: { "Content-Type": "application/json" } },
      ) 
    }
  );
})
```

**Comment by GaryAustin1** [comment]
![image](https://github.com/supabase/supabase/assets/54564956/2ec700a5-b00d-4d6e-95a7-2ec3fdbcd66e)

**Reply by rienheuver** [reply]
This is no longer mentioned in the docs, but I still cannot read files inside my edge function. In my usecase I want to read a file to be used as a template for usage with puppeteer and browserless to generate a pdf

**Reply by brunogonncalves** [reply]
@rienheuver did you find a solution? I'm thinking about saving it in the storage and then read it in the edge function.

**Reply by rienheuver** [reply]
No I did not, we decided to approach it quite differently. We now show a preview of the pdf in the frontend and when a users clicks 'Download', the frontend extracts the entire HTML and CSS and sends that to the edge function. We also thought about storing the template in storage, but that made the DX quite annoying we figured, so we didn't try it.
